### PR TITLE
Dialog to choose from multiple template files per task in wrap

### PR DESCRIPTION
**Pull request by kalisp**
This PR adds possibility to assign multiple Wrap templates for single task, as it is recommended to split Wrap logic into more workfiles for technical reasons.

This functionality is configured in `ayon+settings://wrap/multiple_templates_per_tasks?project=ayon_test` where multiple `steps` could be configured for profile of task name and type.

image

If matching profile is found during start of Wrap, artist is shown simple Workfile app that shows
- list of configured templates in left column
- existing workfiles created from this template
App allows creation (copy of template to workarea) of new workfile, or selection of any existing workfile.
image

This functionality is not required, if single workfile template per task is enough, ordinary `ayon+settings://wrap/workfile_builder?project=ayon_test` should be used.

### CODE REVIEW

File: client/ayon_wrap/hooks/pre_multiple_templates.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                "project_name": project_name,
+                "folder_id": folder_entity["id"],
+                "task_id": task_entity["id"]
+            }
+            workfiles_tool = WorkfilesToolWindow(launch_data=launch_data)
+            workfiles_tool.show()
+            workfiles_tool.raise_()
+            workfiles_tool.activateWindow()
+            workfiles_tool.showNormal()
+
```
Comment (iLLiCiTiT) [inline client/ayon_wrap/hooks/pre_multiple_templates.py]: You don't need to call any of them if you call `exec_()`
```suggestion
```

File: client/ayon_wrap/hooks/pre_multiple_templates.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            workfiles_tool.exec_()
+            workfile_path = os.environ.get("WRAP_WORKFILE_PATH")
+            if not workfile_path or not os.path.exists(workfile_path):
+                raise RuntimeError(f"'{workfile_path} doesn't exist!")
```
Comment (iLLiCiTiT) [inline client/ayon_wrap/hooks/pre_multiple_templates.py]:
```suggestion
                raise ApplicationLaunchFailed(f"'{workfile_path} doesn't exist!")
```

**Comment by kalisp** [comment]
Added documentation PR https://github.com/ynput/ayon-documentation/pull/202

File: client/ayon_wrap/workfiles/control.py [hunk 3]
```diff
@@ -37,6 +37,9 @@ def __init__(self, launch_data):
 
         project_entity = ayon_api.get_project(project_name)
 
+        # template name in Anatomy
+        self.wrap_template_name = "wrap_multi"
```
Comment (iLLiCiTiT) [inline client/ayon_wrap/workfiles/control.py]: If you've already change it to use anatomy then use `get_workfile_template_key` to determine which should be used.

File: client/ayon_wrap/workfiles/widgets/window.py [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        self._fill_templates()
+
+    def _fill_templates(self):
+        template_names = self._controller.get_template_names()
+        templates_widget = self._templates_widget
+        existing_items = {}
+        for idx in templates_widget.count():
+            item = templates_widget.item(idx)
+            if not item:
+                break
```
Comment (iLLiCiTiT) [inline client/ayon_wrap/workfiles/widgets/window.py]:
```suggestion
        for idx in range(templates_widget.count()):
            item = templates_widget.item(idx)
```

File: client/ayon_wrap/workfiles/control.py (hunk 3 above)
Comment (kalisp) [reply client/ayon_wrap/workfiles/control.py]: This workflow is applicable only for specific profiles, to be able to use Anatomy, I would need to duplicate that logic there.

Comment (iLLiCiTiT) [reply client/ayon_wrap/workfiles/control.py]: I don't understand, you are using anatomy, but also hardcoding template name... You should not hardcode template name, you should use whatever is defined in workfile settings, and studio should add the template (they should be able to name it whatever they want) and add it to profiles for `wrap` host.

**Review by iLLiCiTiT — APPROVED** [review]

**Pull request merged**